scripts/masterCron.ts

```diff
@@ -0,0 +1,272 @@
+      log(`${method} (${vaultName}): ${tx.hash}`);
+
+      if (method === "commitAndClose") {
+        returnData[0] = (await vault.delay()).mul(1000);
+      } else if (method === "rollToNextOption") {
+        const receipt = await tx.wait();
+        returnData.push(
+          hexStripZeros(receipt["logs"][15]["topics"][1]).toString().slice(2)
+        );
+      }
+    } catch (error) {
+      await log(
+        `@everyone ${method} (${vaultName}): failed with error ${error}`
+      );
+    }
+  }
+
+  return returnData;
+}
+
+async function masterJob() {
+  const vaultArtifact = await hre.artifacts.readArtifact("RibbonThetaVault");
+  const gnosisArtifact = await hre.artifacts.readArtifact("IGnosisAuction");
+
+  const gnosisAuction = new ethers.Contract(
+    GNOSIS_EASY_AUCTION,
+    gnosisArtifact.abi,
+    provider
+  );
+
+  // Master Cron Job
+  //
+  // 1. commitAndClose
+  let delayBeforeRoll = await runTX(
+    vaultArtifact.abi,
+    provider,
+    signer,
+    network,
+    "commitAndClose"
+  );
+  // 2. wait an hour (timelock period)
+  await sleep(delayBeforeRoll[0]);
+  // 3. rollToNextOption
+  let auctionCounters = await runTX(
+    vaultArtifact.abi,
+    provider,
+    signer,
+    network,
+    "rollToNextOption"
+  );
+  // 4. wait for auctions to close
+  await waitForAuctionClose(auctionCounters, gnosisAuction);
+  // 5. settleAuction
+  await settleAuctions(
+    gnosisAuction,
+    provider,
+    signer,
+    network,
+    auctionCounters
+  );
+  // 6. if otokens left to burn: burnRemainingOTokens
+  await runTX(
+    vaultArtifact.abi,
+    provider,
+    signer,
+    network,
+    "burnRemainingOTokens"
+  );
+  // 7. wait approximately a week
+}
+
+async function volOracleJob() {
+  const volOracleArtifact = await hre.artifacts.readArtifact("VolOracle");
+
+  const volOracle = new ethers.Contract(
+    VOL_ORACLE,
+    volOracleArtifact.abi,
+    provider
+  );
+
+  for (let univ3poolName in deployments[network].univ3pools) {
+    volOracle.commit(deployments[network].univ3pools[univ3poolName]);
+  }
+}
+
+function run() {
+  client.on("ready", () => {
+    console.log(`Logged in as ${client.user.tag}!`);
+    client.user.setPresence({
+      activity: {
+        name: "vault status",
+        type: "WATCHING",
+      },
+      status: "idle",
+    });
+  });
+
+  client.login(process.env.DISCORD_TOKEN);
+
+  //Atlantic/Reykjavik corresponds to UTC
+
+  // 0 0 10 * * 5 = 10am UTC on Fridays. https://crontab.guru/ is a friend
+  const MASTER_JOB_CRON = "0 0 10 * * 5";
+  const VOL_ORACLE_CRON = `* * */${(await volOracle.period())
+    .div(3600)
+    .toString()} * * *`;
+  const CLOSEST_VALID_TIME =
+    (await getTopOfPeriod(provider, await volOracle.period())) * 1000;
+
+  var masterJob = new CronJob(
+    MASTER_JOB_CRON,
+    async function () {
+      await masterJob();
+    },
+    null,
+    true,
+    "Atlantic/Reykjavik"
+  );
+
+  var volOracleJob = new CronJob(
+    new Date(CLOSEST_VALID_TIME),
```

Comment:
cron job within a cron job. first we wait until the first valid timestamp from now where we can call commit. then from that point on we do it every 12 hours, or however long the period is